Tests/InternxtSwiftCoreTests/Services/Network/EncryptTests.swift

```diff
@@ -73,5 +73,127 @@ final class EncryptTests: XCTestCase {
+                   
+        let expectedHexEncryptedResult = "655f26c07d7a555bdaaa56c60112b892ff1063f6790fb5c7588b4ace569acde445e428e0a068ded0a9ec5e6c613c01d4d029394f11"
+        
+        let chunkSizeInBytes = 10
+        
+        var encryptedChunks: [Data] = []
+        let input = InputStream(data: data)
+        try await sut.encryptFileIntoChunks(
+            chunkSizeInBytes: chunkSizeInBytes,
+            totalBytes: data.count,
+            inputStream: input,
+            key: self.cryptoUtils.hexStringToBytes(hexKey),
+            iv: self.cryptoUtils.hexStringToBytes(hexIv)
+        ) {encryptedChunk in
+            encryptedChunks.append(encryptedChunk)
+        }
+        
+        // We should have 5 chunks of 10 bytes each, and 1 chunk of 3 byte = 53 bytes
+        XCTAssertEqual(encryptedChunks[0].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[1].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[2].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[3].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[4].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[5].count, 3)
+        
+        // Check the result
+        var totalEncryptedData: Data = Data()
+        encryptedChunks.forEach{chunk in
+            
+            totalEncryptedData.append(chunk)
+        }
+        
+        let totalBytes = data.count
+        
+        XCTAssertEqual(totalEncryptedData.count, totalBytes)
+        XCTAssertEqual(expectedHexEncryptedResult, totalEncryptedData.toHexString())
+        
+        
+    }
+    
+    
+    func testEncryptStringUsingExactChunks() async throws {
+        
+        let hexIv = "d5e2083f9a5394d1d2414bd520dd25e3"
+        let hexKey = "fdecbc03e63b2433ab750284f4413ec2220eb3c3cea8a87cdddc421550ca9e0f"
+        
+        let data = "PASSWORDpassword123123!!Password123123!!".data(using: .utf8)!
+                   
+        let expectedHexEncryptedResult = "446910f55d50612cdef91687454ff8c5fc1121b02e5ae5916d8c48ca47c58fb044ef22e8ab62cec3"
+        
+        let chunkSizeInBytes = 10
+        
+        var encryptedChunks: [Data] = []
+        let input = InputStream(data: data)
+        try await sut.encryptFileIntoChunks(
+            chunkSizeInBytes: chunkSizeInBytes,
+            totalBytes: data.count,
+            inputStream: input,
+            key: self.cryptoUtils.hexStringToBytes(hexKey),
+            iv: self.cryptoUtils.hexStringToBytes(hexIv)
+        ) {encryptedChunk in
+            encryptedChunks.append(encryptedChunk)
+        }
+        
+        // We should have 4 chunks of 10 bytes each = 40 bytes
+        XCTAssertEqual(encryptedChunks[0].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[1].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[2].count, chunkSizeInBytes)
+        XCTAssertEqual(encryptedChunks[3].count, chunkSizeInBytes)
+        
```

Comment:
got it, nice one, in any case the amount of upload urls that we will have will not match the amount of chunks we have, and it will fail too, but better have it covered indeed